src/main/kotlin/Board.kt

```diff
@@ -0,0 +1,108 @@
+class Board(initialPositions: Map<Char, List<String>> = mapOf()) {
+    private var turn: Int = 0
+    private val boardSpots = arrayOf(charArrayOf(' ', ' ', ' '), charArrayOf(' ', ' ', ' '), charArrayOf(' ', ' ', ' '))
+
+    init {
+        initialPositions.forEach { (xOr0, placements) ->
+            placements.forEach { move ->
+                val (row, col) = parseMove(move)
+                boardSpots[row][col] = xOr0
+            }
+        }
+    }
+
+    fun addMove(move: String) {
+        val (row, col) = parseMove(move)
+
+        if (isZerosTurn(turn)) {
+            boardSpots[row][col] = 'O'
+        } else {
+            boardSpots[row][col] = 'X'
+        }
+        turn++
+    }
+
+    private fun isZerosTurn(turn: Int) = turn % 2 == 0
+
+    fun isOccupied(move: String): Boolean {
+        val (row, col) = parseMove(move)
+        val square = boardSpots[row][col]
+        return square == 'X' || square == '0'
+    }
+
+    fun parseMove(move: String): Pair<Int, Int> {
+        val row = move[0].uppercaseChar().code.minus(65)
+        val col = move[1].digitToInt().minus(1)
+        return Pair(row, col)
+    }
+
+    fun checkPlayerWon(): Boolean {
+        return checkColumns()
+                || checkHorizontal()
+                || checkDiagonalUp()
+                || checkDiagonalDown()
+    }
```

Comment:
This method was initially very large. By breaking it out into sub functions we can focus on each individual kind of check, and use function names as explanatory helpers (and no longer need the comments). This let's us focus our testing on one piece of a time (though I'm still testing at this top level, so I'm not testing implementation). It also makes it easier to scan the code and not worry about checks that are not relevant to whatever you're looking at at the time. Finally, it reduces shared, mutable state since things are now scoped to each of the smaller function.

One other thing to call out is trying to have the same level of abstraction within a function. Had I only abstracted say two of these, it'd be a weird mix of higher and lower level functions, which you want to avoid.